### ISSUE TITLE
insertGet return null if MySQL table contains JSON type field

### ISSUE THREAD

**Issue by RGBvision** [paragonie/easydb#153]
insertGet('table', ['text_field' => 'text', 'json_field' => json_encode([1,2,3])], 'id') will not return 'id'

**Comment by paragonie-security** [comment]
Does your table have an `id` field? The JSON is likely a red herring.

**Comment by RGBvision** [comment]
> Does your table have an `id` field? The JSON is likely a red herring.

Yep. `id` - INT (autoincrement) and primary key. insertGet work as expected if I change JSON data type to TEXT. But in this case I can't use benefits of JSON fields.

**Comment by paragonie-security** [comment]
Circling back to this in 2025:

```php
$db->insertGet(
    'table',
    [
        'text_field' => 'text',
        'json_field' => json_encode([1,2,3])
    ],
    'id'
);
```

This fails with MySQL due to the follow-up SELECT query not quite working for MySQL fields, but you can do this instead:

```php
$db->insertReturnId(
    'table',
    [
        'text_field' => 'text',
        'json_field' => json_encode([1,2,3])
    ]
);
```

**Comment by paragonie-security** [comment]
#164 updates the documentation, tests, and doesn't prevent the other API's use with pgsql.
